### PR TITLE
Add Error and Fatal error alerts.

### PR DESCRIPTION
**Pull request by Mehuge**
Normal errors are displayed in an orange bar at the bottom of the screen.  If there is more than one error the first is displayed and an indicator of how many other errors exist.  Clicking the bar clears that error, showing the next error if there is one.

If the error is fatal, a fatal error dialog is shown that prevents further use of the patcher with a message asking the user to restart the patcher.  If other errors were received before the fatal error, they are also displayed in the dialog.

### CODE REVIEW

**Comment by Mehuge** [comment]
BTW, I left the simulation code in, but commented out.

File: patcher/src/widgets/Controller/components/PatcherError/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    right: 0;
+    height: 100%;
+    width: 30px;
+    pointer-events: none;
+  }
+`;
+
+export interface PatcherErrorProps {
+  errors: PatcherError[];
+  onClear?: any;
```
Comment (Ajackster) [inline patcher/src/widgets/Controller/components/PatcherError/index.tsx]: Declare this as some function type, looks like it's being used as just a `() => void;`

File: patcher/src/widgets/Controller/components/PatcherError/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import styled from 'react-emotion';
+import { PatcherError } from '../../../../services/patcher';
+
+const Alert = styled('div')`
+  overflow: hidden;
+  background-color: orange;
+  color: black;
+  padding: 3px 15px;
+  position: relative;
+  pointer-events: bounding-box;
```
Comment (Ajackster) [inline patcher/src/widgets/Controller/components/PatcherError/index.tsx]: What does this do? I've never seen this before

Comment (Ajackster) [reply patcher/src/widgets/Controller/components/PatcherError/index.tsx]: Errr this isn't actually a request for change, was just asking a question.

Comment (Mehuge) [reply patcher/src/widgets/Controller/components/PatcherError/index.tsx]: Looks like its a chrome thing, and really meant for svg.  Should probably change to : all;

**Comment by Mehuge** [comment]
I have also removed the simulation code, because it's not needed.  To simulate an error do this in the debugger:

1. Search for OnPatcherError and add a conditional breakpoint on the call in onError that says `window.simulateError = handler`
2. Refresh
3. Type `simulateError({ message: 'my error message', fatal: false })` to simulate an error.

**Comment by Ajackster** [comment]
Thanks!

**Pull request merged**